src/Controller/UI/RemotelyHostedDatasetsController.php

```diff
@@ -155,4 +158,48 @@ private function dispatchLogEvent(Dataset $dataset, string $actor)
             'remotelyhosted_update_log'
         );
     }
+
+    /**
+     * Check to see if the remotely-hosted fields are all populated.
+     *
+     * @param DatasetSubmission $datasetSubmission A DatasetSubmission.
+     *
+     * @return array $errors Containing strings of each unpopulated attribute, or empty array for compliant.
+     */
+    public function remotelyHostedPrerequsiteCheck(DatasetSubmission$datasetSubmission) : array
+    {
+        $errors = array();
+
+        if (!$datasetSubmission instanceof DatasetSubmission) {
```

Comment:
I feel this check should be placed in the previous function, because if suppose DatasetSubmission object doesn't exist and this function is called, PHP would throw an error for incorrect parameter type